src/analysis/normed_space/quaternion_exponential.lean

```diff
@@ -0,0 +1,184 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import analysis.quaternion
+import analysis.normed_space.exponential
+import analysis.inner_product_space.pi_L2
+import analysis.special_functions.trigonometric.series
+
+/-!
+# Lemmas about `exp` on `quaternion`s
+
+This file contains results about `exp` on `quaternion ℝ`.
+
+## Main results
+
+* `quaternion.exp_eq`: the general expansion of the quaternion exponential in terms of `real.cos`
+  and `real.sin`.
+* `quaternion.exp_of_re_eq_zero`: the special case when the quaternion has a zero real part.
+
+
+-/
+
+open_locale quaternion nat
+
+-- TODO: move
+lemma div_div_cancel_left' {G₀} [comm_group_with_zero G₀] (a b : G₀) (ha : a ≠ 0) :
+  a / b / a = b⁻¹ :=
+by rw [div_eq_mul_inv, div_eq_mul_inv, mul_right_comm, mul_inv_cancel ha, one_mul]
+
+
+namespace quaternion
+
+section
+variables {α : Type*}
+
+lemma has_sum_coe {f : α → ℝ} {r : ℝ} (h : has_sum f r) :
+  has_sum (λ a, (f a : ℍ[ℝ])) (↑r : ℍ[ℝ]) :=
+by simpa only using h.map (algebra_map ℝ ℍ[ℝ]) (continuous_algebra_map _ _)
+
+@[simp]
+lemma summable_coe_iff  {f : α → ℝ} : summable (λ a, (f a : ℍ[ℝ])) ↔ summable f :=
+by simpa only using summable.map_iff_of_left_inverse (algebra_map ℝ ℍ[ℝ])
+  ({to_fun := re, map_add' := add_re, map_zero' := zero_re} : ℍ[ℝ] →+ ℝ)
+  (continuous_algebra_map _ _) continuous_re coe_re
+
+lemma tsum_coe (f : α → ℝ) : ∑' a, (f a : ℍ[ℝ]) = ↑(∑' a, f a) :=
```

Comment:
The other way around should be `norm_cast`

Reply:
Ah, good point

Reply:
Actually `norm_cast` doesn't care about the direction, it canonicalizes it internally. Added the attribute though.